integrations/nifi/nifi-nar-bundles/nifi-standard-services/nifi-spark-service-bundle/nifi-spark-service/src/main/java/com/thinkbiganalytics/nifi/v2/spark/SparkJobserverService.java

```diff
@@ -0,0 +1,357 @@
+        final List<PropertyDescriptor> props = new ArrayList<>();
+        props.add(JOBSERVER_URL);
+        props.add(SYNC_TIMEOUT);
+        properties = Collections.unmodifiableList(props);
+    }
+
+    @Override
+    protected List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return properties;
+    }
+
+    /**
+     * Configures Spark Jobserver client by creating an instance of the
+     * {@link ISparkJobServerClient} based on configuration provided with
+     * {@link ConfigurationContext}.
+     * <p>
+     * This operation makes no guarantees that the actual connection could be
+     * made since the underlying system may still go off-line during normal
+     * operation of the service.
+     *
+     * A Context Timeout task is also created to delete any contexts which have
+     * been inactive for some time.
+     *
+     * @param context the configuration context
+     * @throws InitializationException if unable to create a Spark Jobserver connection
+     */
+    @OnEnabled
+    public void onConfigured(final ConfigurationContext context) throws InitializationException {
+
+        jobServerUrl = context.getProperty(JOBSERVER_URL).getValue();
+
+        getLogger().info("Creating a new connection to Spark Jobserver: " + jobServerUrl);
+
+        try {
+            client =
+                SparkJobServerClientFactory
+                    .getInstance()
+                    .createSparkJobServerClient(jobServerUrl);
+
+            syncTimeout = context.getProperty(SYNC_TIMEOUT).getValue();
+            setupContextTimeoutTask();
+
+        } catch (Exception ex) {
+            getLogger().error(ex.getMessage());
+        }
+        getLogger().info("Created new connection to Spark Jobserver: " + jobServerUrl);
+    }
+
+    @Override
+    public boolean checkIfContextExists(String contextName) {
+
+        Boolean contextExists = false;
+
+        try {
+            contextExists = client.getContexts().contains(contextName);
+
+            if (!contextExists) {
+                getLogger().info(String.format("Context %s not found on Spark Jobserver %s", contextName, jobServerUrl));
+            } else {
+                getLogger().info(String.format("Context %s exists on Spark Jobserver %s", contextName, jobServerUrl));
+                contextExists = true;
+            }
+
+        } catch (Exception ex) {
+            getLogger().error(ex.getMessage());
+        }
+        return contextExists;
+    }
+
+    @Override
+    public boolean createContext(String contextName, String numExecutors, String memPerNode, String numCPUCores, SparkContextType contextType, int contextTimeout, boolean async) {
+
+        boolean newContext = isNewContext(contextName);
+
+        if (newContext) {
+            Boolean contextRunning = checkIfContextExists(contextName);
+            if (!contextRunning) {
+                try {
+                    Map<String, String> params = new HashMap<String, String>();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace the type specification in this constructor call with the diamond operator ("<>"). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS2293)